### PR TITLE
Use feature gate to control hvpa for kube-apiserver and etcd of seed clusters

### PR DESCRIPTION
**Pull request by ggaurav10**
Use feature gate to control hvpa for kube-apiserver and etcd of seed clusters

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```noteworthy operator
To enable scaling of `kube-apiserver` and `etcd` of shooted seeds via HVPA, enable the feature gate `HVPAForShootedSeed`.
```

### CODE REVIEW

File: pkg/features/features.go [hunk 1]
```diff
@@ -35,4 +35,9 @@ const (
 	// owner @ggaurav10, @amshuman-kr
 	// alpha: v0.1.0
 	HVPA utilfeature.Feature = "HVPA"
+
+	// HVPAForShootedSeed enables simultaneous horizontal and vertical in Soil Clusters.
```
Comment (shreyas-s-rao) [inline pkg/features/features.go]:
```suggestion
	// HVPAForShootedSeed enables simultaneous horizontal and vertical scaling in Soil Clusters.
```

File: pkg/features/features.go [hunk 2]
```diff
@@ -31,8 +31,13 @@ const (
...
-	// HVPA enables simultaneous horizontal and vertical in Seed Clusters.
+	// HVPA enables simultaneous horizontal and vertical scaling in Seed Clusters.
 	// owner @ggaurav10, @amshuman-kr
 	// alpha: v0.1.0
 	HVPA utilfeature.Feature = "HVPA"
+
+	// HVPAForShootedSeed enables simultaneous horizontal and vertical scaling in Soil Clusters.
+	// owner @ggaurav10, @amshuman-kr
+	// alpha: v0.1.0
+	HVPAForShootedSeed utilfeature.Feature = "HVPA-For-ShootedSeed"
```
Comment (rfranzke) [inline pkg/features/features.go]:
```suggestion
	HVPAForShootedSeed utilfeature.Feature = "HVPAForShootedSeed"
```

File: pkg/operation/botanist/controlplane.go [hunk 3]
```diff
@@ -1107,6 +1117,20 @@ func (b *Botanist) DeployETCD(ctx context.Context) error {
...
+			// If HVPA is disabled, delete any HVPA that was already deployed
+			u := &unstructured.Unstructured{}
+			u.SetName(fmt.Sprintf("etcd-%s", role))
+			u.SetNamespace(b.Shoot.SeedNamespace)
+			u.SetGroupVersionKind(schema.GroupVersionKind{
+				Group:   "autoscaling.k8s.io",
+				Version: "v1alpha1",
+				Kind:    "Hvpa",
+			})
+			if err := b.K8sSeedClient.Client().Delete(context.TODO(), u); client.IgnoreNotFound(err) != nil {
```
Comment (rfranzke) [inline pkg/operation/botanist/controlplane.go]:
```suggestion
			if err := b.K8sSeedClient.Client().Delete(ctx, u); client.IgnoreNotFound(err) != nil {
```

**Review by rfranzke — CHANGES_REQUESTED** [review]

**Comment by rfranzke** [comment]
Please add a release note and squash commits.

File: pkg/features/features.go (hunk 1 above)
Comment (rfranzke) [reply pkg/features/features.go]: Don't call it "soil", that's no official term. Use "shooted seed".

**Review by rfranzke — APPROVED** [review]
/lgtm

File: pkg/operation/botanist/controlplane.go [hunk 4]
```diff
@@ -917,6 +909,19 @@ func (b *Botanist) DeployKubeAPIServer() error {
...
+		// If HVPA is disabled, delete any HVPA that was already deployed
+		u := &unstructured.Unstructured{}
+		u.SetName(v1alpha1constants.DeploymentNameKubeAPIServer)
+		u.SetNamespace(b.Shoot.SeedNamespace)
+		u.SetGroupVersionKind(schema.GroupVersionKind{
+			Group:   "autoscaling.k8s.io",
+			Version: "v1alpha1",
+			Kind:    "Hvpa",
+		})
+		if err := b.K8sSeedClient.Client().Delete(context.TODO(), u); client.IgnoreNotFound(err) != nil {
```
Comment (ialidzhikov) [inline pkg/operation/botanist/controlplane.go:922]: I am running with

```yaml
featureGates:
  HVPA: false
  HVPAForShootedSeed: false
```

And I get 
```
lastOperation:
    description: 'Flow "Shoot cluster reconciliation" encountered task errors: [task
      "Deploying main and events etcd" failed: retry failed with context deadline
      exceeded, last error: no matches for kind "Hvpa" in version "autoscaling.k8s.io/v1alpha1"]
      Operation will be retried.'
```

Can we make possible running gcm when these features gates are off?

/cc @ggaurav10

Comment (ggaurav10) [reply pkg/operation/botanist/controlplane.go:922]: thanks @ialidzhikov for bringing this to notice. Fixed in #1678

**Pull request merged**